python/ray/tune/checkpoint_manager.py

```diff
@@ -23,10 +24,11 @@ class Checkpoint:
     MEMORY = "memory"
     PERSISTENT = "persistent"
 
-    def __init__(self, storage, value, result=None):
+    def __init__(self, storage, value, result=None, node_ip=None):
         self.storage = storage
         self.value = value
         self.result = result or {}
+        self.node_ip = node_ip or result.get(NODE_IP, None)
```

Comment:
```suggestion
        self.node_ip = node_ip or (result.get(NODE_IP, None) if result else None)
```
Otherwise an exception is raised before the first step

Reply:
Ah thanks, I'll make it `self.result.get()` instead, which should work